libdevcore/Common.h

```diff
@@ -284,7 +289,11 @@ class Timer
 #define DEV_TIMED_FUNCTION DEV_TIMED_SCOPE(__PRETTY_FUNCTION__)
 #endif
 
+#ifdef __MIC__
+#define DEV_TIMED_ABOVE(S, MS) for (::std::pair<dev::TimerHelper, bool> __eth_t(::dev::TimerHelper(S, MS), true); __eth_t.second; __eth_t.second = false)
```

Comment:
Is the 765 warning suppression above for this line, or for a similar one in a boost header?
For this local problem, the fix is to introduce a space, like so ... 

    ::std::pair< ::dev::TimerHelper